### PR TITLE
Fixes #25844 - 1.21 installer migrations

### PR DESCRIPTION
**Pull request by ekohl**
This is not yet done because I need to go over more modules.

### CODE REVIEW

File: config/foreman.migrations/20190111180118_delete_removed_settings.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+DELETED = {
+  'foreman': [
+    'custom_repo',
+    'dynflow_in_core',
+    'email_conf',
+    'email_config_method',
+    'email_source',
+    'puppet_home',
+    'puppet_ssldir',
+  ],
```
Comment (tbrisker) [inline config/foreman.migrations/20190111180118_delete_removed_settings.rb:10]: Should we also add `authentication`, `locations_enabled` and `organizations_enabled` here?

Comment (ekohl) [reply config/foreman.migrations/20190111180118_delete_removed_settings.rb:10]: They should be in the clear section since the parameters still exist

**Comment by ekohl** [comment]
Turns out my migration was bad. It was using symbols instead of strings which didn't actually delete the answers. Updated now and retesting.

**Comment by ekohl** [comment]
Katello upgrade succeeds. Merging this now and we can see how to deal with other issues that might show up.

**Pull request merged**